api/v1alpha1/amaltheasession_types.go

```diff
@@ -89,6 +89,10 @@ type Session struct {
 	// +optional
 	// +kubebuilder:default:={}
 	Storage Storage `json:"storage,omitempty"`
+	// +optional
+	// +kubebuilder:default:="500Mi"
+	// Size of /dev/shm
+	ShmSize resource.Quantity `json:"shmSize,omitempty"`
```

Comment:
I thought the recommendation was that optional values are pointers. I know we dont follow that everywhere but we can start. I wish I had seen that api convention doc a bit earlier from when we started working on the new operator.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
I had in mind that SHM would be a "mandatory" item hence the use of a value. But following the discussion below it makes even more sense to have it as a pointer.

I'll update that.